docs/src/single-validator-manager.md

```diff
@@ -0,0 +1,122 @@
+costs 1.0028288 SOL. The user receives pool tokens for this deposit.
+
+After the SOL deposit, the `AddValidatorToPool` instruction on the pool management
+program allows anyone to add the stake account for that validator.
+
+Once the added stake account is active, anyone may deposit stake accounts.
+
+### Depositing and withdrawing stake
+
+If stakers deposit or withdraw stake accounts from the pool, everything works
+seamlessly. Stakers receive pool tokens for their deposited stake account, and
+may use the pool tokens to withdraw a stake account.
+
+Activated stake accounts on the same validator are all fungible, so there's no
+other economics to worry about.
+
+### Depositing and withdrawing SOL
+
+As mentioned in the fees section, only SOL deposits have a fee, and everything
+else can be done freely.
+
+To maximize returns for pool depositors, the program allows anyone to maximally
+deploy the reserve to the validator, with important limits.
+
+Single-validator stake pools are an option for small stakers to earn inflation
+rewards by staking their SOL, but if stakers have less than the minimum delegation
+amount, they cannot withdraw stake accounts, so they need a way to withdraw SOL
+from the reserve.
+
+To give a way out for these users, the reserve aims to maintain the minimum delegation
+amount plus stake rent-exemption, or 1.0028288 SOL. If the reserve has less than
+this amount, any user can call `DecreaseValidatorStake` to decrease the minimum
+delegation amount plus rent-exemption.
+
+On the other hand, if the reserve receives enough SOL deposits equalling at least
+two times this amount, then anyone can call `IncreaseValidatorStake` on the manager
+program to activate everything in the reserve, minus the minimum delegation plus
+rent-exemption.
+
+### Griefing
+
+The permissionless increase instruction allows an attack to make the pool less
+performant.
+
+For example, at the start of the epoch, a malicious user can deposit enough SOL to get
+just above the limit for increasing, and immediate run the increase. This prevents
+the activation of subsequent SOL deposits until the next epoch.
+
+However, this attack only works for one epoch, since at the start of the next
+epoch, anyone can activate all of the remaining stake. For this reason, SOL
+depositors are charged 2 epochs of rewards in fees. This inflates the value of
+pool tokens, covering for any economic attacks.
+
+With the permissionless decrease instruction, a similar attack is possible.
+
+For example, if the reserve is at its target amount of 1.0028288 SOL, an attacker
+can withdraw 1 lamport from the reserve, then decrease another 1.0028288 SOL.
+
+Question: should the threshold for allowing decreases be 1 / 2 minimum delegation
+in the reserve? That way, the previous attack is more costly, but it means delegators
+with more than 1 / 2 minimum delegation need to wait longer to get out of the pool.
+That might be an acceptable tradeoff.
+
```

Comment:
is there not also a case where someone might plunder the reserve by depositing active stake, then turning around and swap the pool tokens for SOL for an "instant" stake deactivation?

maybe need to lock SOL withdraws for the remainder of the epoch too

Reply:
Actually, this makes it clear that we want to keep the reserve as small as possible, and force people to use stake accounts for withdrawals as much as possible.

> maybe need to lock SOL withdraws for the remainder of the epoch too

What do you mean by this?

Reply:
> Actually, this makes it clear that we want to keep the reserve as small as possible, and force people to use stake accounts for withdrawals as much as possible.

consider the case where a validator with many sub-min-delegation stakers decides to shut down.  they're caught in deactivation hell for several epochs :thinking: 

> > maybe need to lock SOL withdraws for the remainder of the epoch too
> 
> What do you mean by this?

was thinking we could w/d into program owned accounts that can't be redeemed until after the next epoch boundary. so no "instant" deactivation incentive

Reply:
Hm, true. In those cases, I think you'd have to go through a specialized unstaking service or lending protocol to give you more and perform the unstake.

Would the program owned accounts exist at the stake pool level or some other protocol? Another way to implement this would be to create a deactivating stake account that doesn't get rewards through a new instruction like `SplitAndDeactivateNoRewards`.

Reply:
> a specialized unstaking service

what would this look like?

> Would the program owned accounts exist at the stake pool level or some other protocol?

intuitively, the lowest level that makes accommodations for sub-min-delegation stakers. fwiw i'm envisioning something super simple that just holds lamports and has a state like

```rust
RedemptionCooldownAccountState {
    redeem_after_epoch: Epoch,
    redeem_authority: Pubkey,
}
```

> Another way to implement this would be to create a deactivating stake account that doesn't get rewards through a new instruction like SplitAndDeactivateNoRewards.

i'm panicking at the thought of all of the runtime changes already! :sweat_smile:

Reply:
> what would this look like?

something like https://www.unstake.it/en/

> intuitively, the lowest level that makes accommodations for sub-min-delegation stakers

That would be at the stake program :laughing: 

> i'm panicking at the thought of all of the runtime changes already!

I really don't think it would be that bad! But I'm also a hopeless optimist